run_model.py

```diff
@@ -18,67 +21,186 @@
+    val_to_add = analysis_config_entry["factor_start"]
+    step_values = []
+    while val_to_add <= analysis_config_entry["factor_end"]:
+        step_values.append(val_to_add)
+        val_to_add += analysis_config_entry["factor_step"]
+    return step_values
+
+def compare_to_baseline_values(field_to_update, output_data, baseline_step_value_index):
+    return ['{}%'.format(value/(output_data[field_to_update][baseline_step_value_index]) * 100) for value in output_data[field_to_update]]
 
 if __name__ == '__main__':
+
     parser = argparse.ArgumentParser('CommCare Cluster Model')
-    parser.add_argument('config', help='Path to config file')
+    parser.add_argument('server_config', help='Path to cluster config file')
+    parser.add_argument('-a', '--analysis_config', help='Path to analysis config file', required=False, )
     parser.add_argument('-o', '--output', help='Write output to Excel file at this path.')
     parser.add_argument('-s', '--service', help='Only output data for specific service.')
 
     args = parser.parse_args()
 
     pd.options.display.float_format = '{:.1f}'.format
 
-    config = config_from_path(args.config)
-    usage = generate_usage_data(config)
-    if args.service:
-        config.services = {
-            args.service: config.services[args.service]
-        }
-
-    service_data = generate_service_data(config, usage)
+    if args.analysis_config:
+        analysis_config = dict_config_from_path(args.analysis_config)
+
+        # Get the server config as a dict
+        base_server_config = dict_config_from_path(args.server_config)
+        # Make a list of ClusterConfig objects with all the different modifications specified in the analysis_config file 
+        all_server_config_data = generate_server_configs(base_server_config, analysis_config)
+
+        # Create the dictionary, which will eventually be written to excel
+        output_data = {'Field to Modify': [],
+                       'Subfield to Modify': [],
+                       'Step Value': [],
+                       'RAM (GB)': [],
+                       'CPU (Total Cores)': [],
+                       'SAS Storage (TB)': [],
+                       'SSD Storage (TB)': [],
+                       'Total VMs': [],}
+
+        # Generate server and usage data for each ClusterConfig object
+        for server_config_data in all_server_config_data:
+            server_config = server_config_data['model']
+            usage = generate_usage_data(server_config)
+            service_data = generate_service_data(server_config, usage)
+            summary_data = get_summary_data(server_config, service_data)
+
+            if server_config.summary_dates:
+                summary_dates = server_config.summary_date_vals
+            else:
+                summary_dates = [usage.iloc[-1].name]  # summarize at final date
+
+
+            date_list = list(usage.index.to_series())
+            summary_data = get_summary_data(server_config, service_data)
```

Comment:
repeat of line 106

Reply:
Removed: https://github.com/dimagi/commcare_resource_model/pull/25/commits/ccbf86af0be22a7a7d78b4858ac146bfeafcf7c3